src/StoreKeeper/WooCommerce/B2C/PaymentGateway/PaymentGateway.php

```diff
@@ -279,7 +279,44 @@ public function onReturn()
         wp_redirect($url);
     }
 
-    public function createRefundPayment($orderId, $refundId): void
+    public function createWooCommerceRefund(\WC_Order_Refund $refund, $args)
+    {
+        $orderId = $args['order_id'];
+        $isRefundCreationAllowed = true;
+        $storeKeeperOrderId = get_post_meta($orderId, 'storekeeper_id', true);
+
+        if ($storeKeeperOrderId) {
+            // Refunded dirty means it's just forced to be refunded by BackOffice status
+            $refundedDirty = 'yes' === get_post_meta($orderId, 'refunded_dirty', true);
```

Comment:
why do we need the dirty check, can't we just always check if it storekeeperOrderHasRefundWithoutReturnPayment

Reply:
We can't purely rely on `storekeeperOrderHasRefundWithoutReturnPayment` on this part. Because when the order is imported and the status of WC_Order is changed to `refunded`, it will call the `wc_create_refund` which will then create an unnecessary refund, so we have to check first if it just happened.

Reply:
Agreed to change approach to use static variable instead.